CIPs/cip-37.md

```diff
@@ -14,42 +14,268 @@ created: 2021-01-07
 ## Motivation
 
-Currently, Conflux and Ethereum addresses are very similar, in many cases they are cross-compatible, i.e. some valid Ethereum addresses are valid Conflux addresses and vice versa. This has led many users to lose their funds when transferring them to a Conflux address on Ethereum, particularly during cross-chain operations. A new key type that is incompatible with Ethereum would allow tools like Metamask to prevent such erroneous transactions from being submitted.
+Currently, Conflux and Ethereum addresses are very similar, in many cases they are cross-compatible, i.e. some valid Ethereum addresses are valid Conflux addresses and vice versa. This has led many users to lose their funds when accidentally transferring them to a Conflux address on Ethereum, particularly during cross-chain operations. A new address type that is incompatible with Ethereum would allow tools like Metamask to prevent such erroneous transactions from being submitted to the network.
+
+<!-------------------->
 
 ## Specification
 
-TBD
+*(Parts of this section are based on the Bitcoin Cash [specification](https://github.com/bitcoincashorg/bitcoincash.org/blob/master/spec/cashaddr.md).)*
+
+This section defines conversion between the following address types:
+
+- **conflux-hex-address**: a valid, 20-byte, type-prefixed Conflux hex address, e.g. `0x106d49f8505410eb4e671d51f7d96d2c87807b09`. The way this address is derived from public keys is defined by Equation (1) of the protocol [spec](https://confluxnetwork.org/files/Conflux_Protocol_Specification_20201020.pdf). The string representation of this address can optionally use *mixed-case checksum address encoding* ([EIP-55](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md)).
+
+- **conflux-base32-address**: a network-prefixed Conflux base32-checksum address. The address consists of a network-prefix indicating the network on which this address is valid, a colon (`":"`), and a base32-encoded payload indicating the destination of the address and containing a checksum, e.g. `cfx:0086ujfsa1a11uuecwen3xytdmp8f03v140ypk3mxc`. Optionally, the address can contain a list of key value pairs in the format `key=value` between the network-prefix and the payload, separated by colons, e.g. `cfx:type=user:0086ujfsa1a11uuecwen3xytdmp8f03v140ypk3mxc`
+
+<!-------------------->
+
+### Encoding
+
+INPUT: an `addr` (20-byte conflux-hex-address), a `network-id` (4 bytes)
+
+OUTPUT: a conflux-base32-address
+
+1. **Network-prefix**:
+
+```
+    match network-id:
+        case 1029: "cfx"
+        case 1:    "cfxtest"
+        case n:    "net[n]"
+```
+
+Examples of valid network-prefixes: `"cfx"`, `"cfxtest"`, `"net17"`
+
+Examples of invalid network-prefixes: `"bch"`, `"conflux"`, `"net1"`, `"net1029"`
+
+When presented to users, the prefix may be omitted as it is part of the checksum computation. The checksum ensures that addresses on different networks will remain incompatible, even in the absence of an explicit prefix.
```

Comment:
If this can be omitted to the very end user, all wallets, sdks have to accept both prefixed address and non-prefixed addresses. 
Maybe we can make it mandatory? Or this is out of the scope of this cip and we can standardize the behaviour in application level with another cip?

Reply:
I agree, it's less confusing if we make it consistent. Removed.